### ISSUE TITLE
yamllint breaks import of unchanged existing roles

### ISSUE THREAD

**Issue by dschneller** [ansible/galaxy#824]
##### SUMMARY
Importing existing (unchanged) roles fails with yamllint error due to trailing newline.
This appears to be a new behaviour as of 2018-07-03, because a recent import before that did not exhibit the issue. 

##### STEPS TO REPRODUCE
Import any role where one of the yml files has a trailing empty line. This will break the import with an error.

##### EXPECTED RESULTS
I believe an empty line at the end of a file may warrant a warning, if one's picky, but breaking the import for it seems overly strict. This breaks imports of a number of (probably not just our?) roles, which have (harmless) empty lines at the end of one or more of their files.

##### ACTUAL RESULTS
```
INFO Starting import: task_id=259232, repository=Rheinwerk/ansible-role-readahead_setup
INFO Content search - Looking for file "apb.yml"
INFO Content search - Looking for top level role metadata file
WARNING [role] Value of license has not been set in metadata.
INFO [role] Linting...
ERROR [role] ./defaults/main.yml:6:1: [error] too many blank lines (1 > 0) (empty-lines)
ERROR [role] ./meta/main.yml:14:1: [error] too many blank lines (1 > 0) (empty-lines)
ERROR [role] ./tasks/main.yml:19:1: [error] too many blank lines (1 > 0) (empty-lines)
Task "259232" failed: Lint failed
```

**Comment by cutwater** [comment]
@dschneller Currently we don't distinguish linter warnings and errors. Moreover according to yamllint output, it's considered to be an error. I can't tell if it's something we're going to change in the future.

However the mandatory static analysis is something that should not break existing content at least between releases. So with after `3.0.2` release mandatory linting for roles will be disabled until next major release. It will still produce linting errors to the log output, but will not cause import process failure.
So this should give a required transitional period before next major release, when mandatory linting will be re-enabled again.

**Comment by chouseknecht** [comment]
Marking this as closed, since we're releasing 3.0.2, and the change in #825 should allow the import to complete successfully, even though there are lint warnings/errors.
